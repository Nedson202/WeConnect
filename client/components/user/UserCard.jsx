import React from 'react';
import PropTypes from 'prop-types';
/**
 * @description react stateless component for card template
 * @return {Object} action dispatched by the action creator
 */
export default function UserCard({ user, removeUser, markUserToRemove }) {
  const { username, id } = user;
  return (
    <div className="col-sm-12 col-md-4 col-lg-4">
      <div className="card business-card" style={{ height: '130px' }}>
        <div className="card-header text-center textcase">
          <h5>{username}</h5>
        </div>
        <div className="card-body text-center bg-transparent">
          <button
            className="btn btn-outline-success text-danger"
            data-toggle="modal"
            data-target="#confirmDeleteModal"
            onClick={() => markUserToRemove(id)}
          ><i className="fa fa-trash fa-lg" />
          </button>
        </div>
      </div>

      <div className="modal fade" id="confirmDeleteModal" tabIndex="-1" role="dialog" aria-labelledby="confirmModalTitle" aria-hidden="true">
        <div className="modal-dialog modal-dialog-centered" role="document">
          <div className="modal-content">
            <div className="modal-body">
              <h4>Are you sure?</h4>
            </div>
            <div className="modal-footer">
              <button type="button" className="btn btn-outline-success" data-dismiss="modal">No</button>
              <button type="button" className="btn btn-outline-success" onClick={() => removeUser()} data-dismiss="modal">Yes</button>
            </div>
          </div>
        </div>
      </div>
    </div>
  );
}

UserCard.propTypes = {
  removeUser: PropTypes.func.isRequired,
  markUserToRemove: PropTypes.func.isRequired,
  user: PropTypes.object.isRequired
};
