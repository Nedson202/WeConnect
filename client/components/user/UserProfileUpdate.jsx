import React from 'react';
import PropTypes from 'prop-types';
import ImageUpload from '../ImageUpload/ImageUpload';

/**
 * @class UserProfileUpdate
 *
 * @extends {Component}
 */
const UserProfileUpdate = ({
  state, onChange, onSubmit, user, uploadToCloudinary, onImageChange
}) => {
  const { username, email } = state;
  return (
    <div>
      <div className="modal fade" id="updateModal" tabIndex="-1" role="dialog" aria-labelledby="confirmModalTitle" aria-hidden="true">
        <div className="modal-dialog modal-dialog-centered" role="document">
          <div className="modal-content">
            <div className="modal-header">
              <h4>Edit profile</h4>
              <button type="button" className="close" data-dismiss="modal" aria-label="Close" id="closeProfileModal">
                <span aria-hidden="true">&times;</span>
              </button>
            </div>
            <div className="modal-body">
              <form>
                <div className="row">
                  <div className="form-group col-lg-12">
                    <label id="upload-form-label">Username</label>
                    <input
                      value={username}
                      onChange={onChange}
                      type="text"
                      name="username"
                      className="form-control"
                      id="control-label"
                      placeholder="username"
                    />
                  </div>
                  <div className="form-group col-lg-12">
                    <label id="upload-form-label">Email</label>
                    <input
                      value={email}
                      onChange={onChange}
                      type="text"
                      name="email"
                      className="form-control"
                      id="control-label"
                      placeholder="email"
                    />
                  </div>
                  <ImageUpload
                    onImageChange={onImageChange}
                    uploadToCloudinary={uploadToCloudinary}
                    user={user}
                    state={state}
                  />
                </div>
              </form>
            </div>
            <div className="modal-footer">
              <button type="button" className="btn btn-outline-success" data-dismiss="modal">No</button>
              <button
                className="btn btn-outline-success"
                onClick={onSubmit}
                id="profile-update-button"
              >Upload
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
  );
};

UserProfileUpdate.propTypes = {
  onChange: PropTypes.func.isRequired,
  onImageChange: PropTypes.func.isRequired,
  onSubmit: PropTypes.func.isRequired,
  state: PropTypes.object.isRequired,
  uploadToCloudinary: PropTypes.func.isRequired,
  user: PropTypes.object.isRequired,
};

export default UserProfileUpdate;
