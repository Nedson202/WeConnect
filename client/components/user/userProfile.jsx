import React from 'react';
import Navbar from '../Navbar';
import Footer from '../Footer';
import ImageUploader from '../ImageUpload/ImageUpload'
import '../../index.scss';

const UserProfile = () => {
  return (
    <div>
      <Navbar />

      <div className="container">
        <div className="row col-lg-10 offset-lg-1">
          <div className="col-lg-2">
            <ImageUploader />
          </div>
        </div>
      </div>

      <Footer />
    </div>
  );
}

export default UserProfile;
