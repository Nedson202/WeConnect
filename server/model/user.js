const users = [
  {
    id: 1,
    username: 'michael',
    email: 'mike@smith.lin',
    password: 'asynch',
  },
  {
    id: 2,
    username: 'alfred',
    email: 'alfred@gmail.com',
    password: 'fadaha',
  }
];

export default users;
