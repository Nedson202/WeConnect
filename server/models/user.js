

module.exports = (sequelize, DataTypes) => {
  const User = sequelize.define('User', {
    username: {
      type: DataTypes.STRING,
      unique: {
        args: true,
        msg: 'Username is already taken'
      }
    },
    email: {
      type: DataTypes.STRING,
      unique: {
        args: true,
        msg: 'Email is already taken'
      }
    },
    password: {
      type: DataTypes.STRING,
      allowNull: false
    }
  });

  User.associate = (models) => {
    User.hasMany(models.Business, {
      foreignKey: 'userId',
      as: 'users',
      onDelete: 'CASCADE'
    });
  };

  return User;
};
