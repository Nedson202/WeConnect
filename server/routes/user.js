import Auth from '../controller/user';
import validator from '../middlewares/validator';
import errorHandler from '../middlewares/error-handler';

export default (route) => {
  route.post('/api/v1/auth/signup', validator.userSignup, errorHandler, Auth.createUser);
  route.post('/api/v1/auth/login', validator.userLogin, errorHandler, Auth.logUser);
};
