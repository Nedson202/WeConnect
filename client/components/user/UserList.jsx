import React from 'react';
import PropTypes from 'prop-types';
import UserCard from './UserCard';

/**
   * @description Creates users list.
   * @param {prop} users
   * @param {prop} deleteUser
   * @returns {object} JSX object
   * @memberof UserList
   */
const UserList = ({ users, removeUser, markUserToRemove }) => {
  const noUser = (
    <h4 className="text-center">No user registered yet</h4>
  );

  const userList = (
    <div className="row">
      { users.map(user =>
        (<UserCard
          user={user}
          removeUser={removeUser}
          markUserToRemove={markUserToRemove}
          key={user.id}
        />))}
    </div>
  );

  return (
    <div>
      <div className="text-center">
        {users.length === 0 ? null :
        <h4>Displaying user list</h4>
        }
      </div>
      <div className="container card-container">
        {users.length === 0 ? noUser : userList}
      </div>
    </div>
  );
};

UserList.propTypes = {
  removeUser: PropTypes.func.isRequired,
  markUserToRemove: PropTypes.func.isRequired,
  users: PropTypes.array.isRequired
};

export default UserList;
