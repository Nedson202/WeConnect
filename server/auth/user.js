import bcrypt from 'bcrypt';
import jwt from 'jsonwebtoken';
import models from '../models/index';
import config from '../config/config';

const Users = models.User;
/**
 *Represents user authentication
 *@class
 */
class Auth {
  /**
      *
      *@param {any} req - request value
      *@param {any} res - response value
      *@return {json} response object gotten
      *@memberof Auth
    */
  static createUser(req, res) {
    const { username, email, password } = req.body;

    Users.create({
      username: username.toLowerCase(),
      email: email.toLowerCase(),
      password: password.toLowerCase()
    })
      .then((user) => {
        const passwordHash = bcrypt.hashSync(user.password, 10);
        user.update({
          password: passwordHash
        });
        return res.status(201).json({
          message: 'Signup successful',
          error: false,
        });
      }).catch(error => res.status(409).json({
        message: error.errors[0].message,
        error: true
      }));
  }

  /**
      *
      *@param {any} req - request value
      *@param {any} res - response value
      *@return {json} response object goten
      *@memberof Auth
    */
  static logUser(req, res) {
    const { username } = req.body;

    Users.findOne({
      where: {
        username: username.toLowerCase()
      }
    })
      .then((user) => {
        if (!(user && bcrypt.compareSync((req.body.password).toLowerCase(), user.password))) {
          return res.status(401).json({
            message: 'Unathorised, check your username or password',
            error: true
          });
        }

        const token = jwt.sign(
          {
            userId: user.id,
            username,
            email: user.email
          },
          config.secretkey,
          {
            expiresIn: '10h'
          }
        );

        return res.status(200).json({
          message: 'login successful',
          tokenMessage: 'Generated token expires in 10 hours time',
          error: false,
          token
        });
      });
  }
}

export default Auth;
