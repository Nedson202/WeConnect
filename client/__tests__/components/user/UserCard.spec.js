import React from 'react';
import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import toJSON from 'enzyme-to-json';
import UserCard from '../../../components/user/UserCard';
import user from '../../__mockData__/userData';

configure({ adapter: new Adapter() });

let props;
const setup = () => {
  props = {
    user,
    removeUser: jest.fn(),
    markUserToRemove: jest.fn()
  };

  return shallow(<UserCard {...props} />);
};


describe('Component: UserCard', () => {
  it('should render', () => {
    const wrapper = setup();
    expect(wrapper.find('div').length).toBe(9);
    expect(toJSON(wrapper)).toMatchSnapshot();
    expect(wrapper.find('h4').length).toBe(1);
    expect(wrapper.find('h5').length).toBe(1);
    expect(wrapper.find('button').length).toBe(3);
    expect(wrapper.find('.delete-button').simulate('click'));
    expect(wrapper.find('.dismiss-button').simulate('click'));
    expect(wrapper.find('.remove-user').simulate('click'));
    expect(wrapper.length).toBe(1);
  }) 
});