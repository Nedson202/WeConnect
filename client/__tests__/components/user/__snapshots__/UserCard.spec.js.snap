// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: UserCard should render 1`] = `
<div
  className="col-sm-12 col-md-4 col-lg-4"
>
  <div
    className="card business-card"
    style={
      Object {
        "height": "130px",
      }
    }
  >
    <div
      className="card-header text-center textcase"
    >
      <h5 />
    </div>
    <div
      className="card-body text-center bg-transparent"
    >
      <button
        className="btn btn-outline-success text-danger delete-button"
        data-target="#confirmDeleteModal"
        data-toggle="modal"
        onClick={[Function]}
      >
        <i
          className="fa fa-trash fa-lg"
        />
      </button>
    </div>
  </div>
  <div
    aria-hidden="true"
    aria-labelledby="confirmModalTitle"
    className="modal fade"
    id="confirmDeleteModal"
    role="dialog"
    tabIndex="-1"
  >
    <div
      className="modal-dialog modal-dialog-centered"
      role="document"
    >
      <div
        className="modal-content"
      >
        <div
          className="modal-body"
        >
          <h4>
            Are you sure?
          </h4>
        </div>
        <div
          className="modal-footer"
        >
          <button
            className="btn btn-outline-success dismiss-button"
            data-dismiss="modal"
            type="button"
          >
            No
          </button>
          <button
            className="btn btn-outline-success remove-user"
            data-dismiss="modal"
            onClick={[Function]}
            type="button"
          >
            Yes
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;
